### PR TITLE
Feature/gsye 650 fix

### PR DESCRIPTION
**Pull request by spyrostz**

### CODE REVIEW

File: gsy_framework/sim_results/kpi.py [hunk 1]
```diff
@@ -140,22 +147,24 @@ def _dissipate_infinite_consumption(self, trade: Dict):
...
+                    self._accumulate_infinite_consumption(trade)
+                    self._dissipate_infinite_consumption(trade)
 
     def update_area_kpi(self, area_dict: Dict, core_stats: Dict):
         """Update kpi after every market cycle"""
         self.total_energy_demanded_wh = 0
+
         self._accumulate_total_energy_demanded(area_dict, core_stats)
-        self._accumulate_energy_trace(core_stats)
+        self._accumulate_energy_trace(core_stats, "P2P" in area_dict["name"])
```
Comment (hannesdiedrich) [inline gsy_framework/sim_results/kpi.py:167]: Do we really want to add this solution permanently?
If yes, we should add a more unique string IMO.
e.g. `##P2P##` What do you think?

File: gsy_framework/sim_results/kpi.py [hunk 2]
```diff
@@ -49,11 +49,13 @@ def __init__(self):
 
     def accumulate_devices(self, area_dict: Dict):
         """Accumulate all child devices with respective types"""
+        if "DH" in area_dict["name"]:
+            return
         for child in area_dict["children"]:
-            if is_producer_node_type(child):
+            if is_producer_node_type(child) or is_heatpump_node_type(child):
```
Comment (hannesdiedrich) [inline gsy_framework/sim_results/kpi.py:55]: why is the heat pump also a producer?

**Review by hannesdiedrich — COMMENTED** [review]
Two comments, LGTM once addressed

File: gsy_framework/sim_results/kpi.py (hunk 2 above)
Comment (spyrostz) [reply gsy_framework/sim_results/kpi.py:55]: Should not be and good catch, thanks!

File: gsy_framework/sim_results/kpi.py (hunk 1 above)
Comment (spyrostz) [reply gsy_framework/sim_results/kpi.py:167]: Let me think of it a bit more and adapt it. Same goes for the "DH" string for distinguishing district heating.

**Comment by hannesdiedrich** [comment]
@spyrostz We should also discuss whether the distinguishing between P2P and DH should end up in our codebase (whether we should push this)

**Pull request merged**